Update some CICE variable names to clarify grid

apcraig opened the pull request:
## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Update some CICE variable names to clarify grid
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    All tests bit-for-bit, full test suites run on cheyenne, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#3079979f165ab96f73fb1b8e51d77bd027bce39e
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [X] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [X] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [X] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [X] Please provide any additional information or relevant details below:

Partly addresses several items in #660 

-Rename several variables to make compatible with C-grid names,
  - strairxU, strairyU, strtltxU, strtltyU, strintxU, strintyU,
  - taubxU, taubyU, strocnxU, strocnyU, fmU, TbU,
  - waterxU, wateryU, forcexU, forceyU, aiU
- Move iceumask, icenmask, iceemask from ice_flux to ice_grid
- Make dyn_prep2, stepu, stepuv_CD, stepv_C, implicit_solver, and anderson_solver argument names a little more generic/specific
- Inline boxslotcyl velocity calculation

apcraig commented:
iceUmask is a very reasonable suggestion.  I decided to just leave it as is for now, but if others prefer iceUmask, iceTmask, iceEmask, iceNmask, I'm happy to make that change throughout.

eclare108213 commented on cicecore/cicedynB/dynamics/ice_dyn_shared.F90:
```diff
@@ -497,34 +497,34 @@ subroutine dyn_prep2 (nx_block,   ny_block,   &
       integer (kind=int_kind), intent(out) :: &
          icellt  , & ! no. of cells where icetmask = 1
-         icellu      ! no. of cells where iceumask = 1
+         icellx      ! no. of cells where icexmask = 1
```
I would probably have simply left the x's out of all these generic variables to avoid confusion with variables that use x as a direction, although this approach is nicely parallel with the t values.  Note these x's are not capitalized, while others elsewhere are.

dabail10 reviewed (approved):
This looks great.

eclare108213 commented on cicecore/cicedynB/infrastructure/ice_grid.F90:
```diff
@@ -173,8 +173,7 @@ module ice_grid
-      logical (kind=log_kind), &
-         dimension (:,:,:), allocatable, public :: &
+      logical (kind=log_kind), dimension (:,:,:), allocatable, public :: &
          tmask  , & ! land/boundary mask, thickness (T-cell)
```
As long as you're changing grid variable names, you might as well make them all consistent.

eclare108213 commented on cicecore/cicedynB/infrastructure/ice_grid.F90:
```diff
@@ -183,7 +182,12 @@ module ice_grid
+         icenmask, &   ! ice extent mask (N-cell)
+         iceemask      ! ice extent mask (E-cell)
+
+real (kind=dbl_kind), dimension (:,:,:), allocatable, public :: &
```
Is this column misaligned?

eclare108213 reviewed (approved):
This all looks fine to me, with a couple of suggestions.

apcraig commented:
I updated the indentation and modified several more variables names in ice_dyn_shared to better indicate the grid by capitalizing the grid "letter".  I also kept the "x", but generally made it "X".  I am a little reluctant to change some of the primary model variable names (i.e. tmask to maskT or Tmask) without a bit more discussion/thought.  I think this is ready for another review.

eclare108213 reviewed (approved):
Terrific, thank you.  Since Fortran doesn't recognize capitalization this shouldn't make any difference to the code, but it makes a huge improvement in readability for humans.
